### ISSUE TITLE
Making navbar width consistent with the main layout on large displays

### ISSUE THREAD

**Issue by jetxr** [facebook/docusaurus#7562]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Description

- On large displays, the navbar's width isn't matched to the main content, footer, and it expands to the edges of the screen making it easy to miss the navbar's contents. (demonstrating the issue on docusaurus.io Ref: [link](#issue))

- However, if `max-width` and `justify-content` are applied, it makes the navbar more consistent with the rest of the design.

- I've added the following to my `custom.css`, and it seems to be working

```pcss
.navbar {
	justify-content: center;
}

.navbar__inner {
	max-width: var(--ifm-container-width-xl);
}
```

- Should this be Docusaurus' default behavior?

---

### Issue
![docusaurus(iMac) - Issue](https://user-images.githubusercontent.com/13171853/172013648-32f0791c-941d-4e24-b0ff-79642094c1f9.png)

---
### Suggested Change
![docusaurus(iMac) - Suggestion](https://user-images.githubusercontent.com/13171853/172017505-040559e3-2505-4873-8b7c-969beba73c9a.png)

### Motivation

- On large displays, with main body having a max width, it's looks a bit out of place to have the menu items at the edges, and not have the navbar as the same width as the rest of the content.

- When the nabar is the same max width as the content area, you don't need to scan the entire width of the screen. The entire page stays a vertical scannable area. For ex. https://developer.apple.com/documentation/Xcode/Xcode-Cloud

### Self-service

- [ ] I'd be willing to contribute this feature to Docusaurus myself.

**Comment by Josh-Cena** [comment]
Agree, feel free to send a PR!

**Comment by jetxr** [comment]
Yes, makes sense to apply it to screens > 1440px. Could use a media query to do that. Will try and send a PR this week.

**Comment by Josh-Cena** [comment]
It's a bit awkward that we only have 996px as the only cutoff currently. We should have a more nuanced screen width system, like xl/lg/md etc... For example, we would want to display icons-only on medium-sized screens (e.g. ipads) (ref #7566)

**Comment by LichLord91** [comment]
Added this to my own  website, however I noticed the Sidebar in docs (at least when collapsible) sticks to the left as well and in @jetxr  example of the Apple site it remains centered with the rest of the content. Any suggestions in changing this behavior?


![image](https://user-images.githubusercontent.com/8435580/172729580-c38c1bf1-1153-4037-90b4-655533e8aee8.png)

![image](https://user-images.githubusercontent.com/8435580/172729685-28125ab7-c3c9-40e7-9348-85468ff26120.png)

**Comment by jetxr** [comment]
Adding a `max-width` and `margin: auto` to `main-wrapper` seemed to work. Not sure if it's the best way to do it. Also, new to Docusaurus, so I don't know if this will have an effect on something else. Maintainers can confirm.

![Screen Shot 2022-06-09 at 10 17 28 AM](https://user-images.githubusercontent.com/13171853/172768138-2a5ab710-6f75-45ab-bff1-b3c1d261b975.png)

To be honest, sidebar sticking to the edge of a large screen, while having the navbar centered does looks odd.  So on docs pages, if the navbar has to be centered, the sidebars should be as well.

On non-docs pages, the centered navbar might be preferable for some. But on the docs pages, both the options (full-width navbar + left-aligned sidebar _vs_ centered page) have their uses.

Might be a better idea to make this configurable, and part of a larger task?

**Comment by Josh-Cena** [comment]
We'd rather offer a good default and let users override with custom CSS.

**Comment by jetxr** [comment]
Makes sense. Adding a couple of lines of CSS does the work.

I do however think having the navbar centered for large-screens on non-docs pages is a better default.

@Josh-Cena leaving it up to you if we can keep this issue open. As such, there is a solution.

**Comment by Josh-Cena** [comment]
The site with the closest layout that comes to mind is https://www.typescriptlang.org/. Here's what they got:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/55398995/172776608-1497a353-1c37-427f-91f1-2067029f3dbe.png">

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/55398995/172776629-60a0776f-ef12-4bc7-904f-dd075e1e1ad3.png">

The navbar grows infinitely, but the main content has max-width. That's _very_ close to what we do.

In Docusaurus v1, the navbar has a constrained width.

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/55398995/172776951-a97a2126-41f7-4af1-8d1a-c39f20ab6443.png">

Related to #4347. @orta ideas on how the navbar should grow?

**Comment by orta** [comment]
Alas "it depends" there are a few constraints to consider with this decision (I'll describe them from TS' side)

1. The navbar being consistent feels pretty important as its a primary way to get around IMO
2. Different pages need different width constraints
    - Long bodies of text gets much harder to read at wider widths, 
    - Pages like the playground want to be full bleed at whatever side

For the TS site, I opted to have a consistent left nav rather than having it changing because the playground represents a large amount of the site traffic.

Looking at Jest's docs, there isn't really a page which **needs** a full bleed layout - the design conflict comes from the nav column which always aligns left edge and the content which is max width and centered. If the navigation column moves in with the center, then I'd say its a pretty easy win for team 'center the top nav' 

![image](https://user-images.githubusercontent.com/49038/172779510-b93fe737-de36-4160-a4f9-eab61327ba66.png)

If keeping that permanently right aligns is a preferred, then I think it might still feel better to keep the nav also right + left aligned

**Comment by jetxr** [comment]
Just a thought.. would it be ideal to set the max width of the entire site to 1730px? (16" MBP)

Anything wider would probably be an external display/very large display, and it's not too comfortable to scan content horizontally at that width.

It would:
- Make the navbar be consistent on all pages, and keep sidebar navigation close/closer to content
- Suits a documentation site, which will mostly be readable content
- Can be turned off for the whole site with global CSS if the user wants to make whole site full bleed
- On pages which need to be 100% full bleed even on super large screens, the max-width can be turned off, and the content can be placed in 'break-out' sections and won't look broken (?)

![Screen Shot 2022-06-09 at 1 23 45 PM](https://user-images.githubusercontent.com/13171853/172795639-ce09897e-2d4e-42d2-877e-f92c7e075026.png)

(TypeScript Playground with full-bleed but constrained navbar)

**Comment by LichLord91** [comment]
Looking at [https://theochu.com](https://theochu.com )and reading the css in dev tools. I like the solution he came up with.  Seems to work great on mobile/full screen and with collapsible button.

```css
@media (min-width: 1416px) {
	.main-wrapper {
		align-self: center;
		max-width: 1400px;
		width: 1400px;
	}
}

.navbar .navbar__inner {
    margin: 0 auto;
    max-width: 1360px;
}
```


![image](https://user-images.githubusercontent.com/8435580/177637928-4e3dfe76-fa90-4473-9f6a-1c5b3e8339aa.png)

**Comment by alanlivio** [comment]
The @jetxr initial proposal worked for me. 
Other method is this [css](https://github.com/facebook/react-native-website/blob/main/website/src/css/customTheme.scss) used by the ReactNative website, for instance, in [this page](https://reactnative.dev/docs/components-and-apis/)

**Comment by rtxa** [comment]
@LichLord91 What if I just want to apply the centered content to the Docs page only? I have some indivual pages that requires the background to expand to 100% width.

This CSS code fixes the issue by applying full content centering only on the Docs page, while leaving the navbar centering enabled on the rest of the pages (Thanks @oyatek for mentioning about `html.docs-doc-page`).
```css
/* This will fix the navbar and sidebar not being centered on large screens */
@media (min-width: 1416px) {
  /* Only apply sidebar centering on Docs page */
  html.docs-doc-page .main-wrapper {
		align-self: center;
		max-width: 1400px;
		width: 1400px;
	}
}

.navbar .navbar__inner {
    margin: 0 auto;
    max-width: 1360px;
}
```

**Comment by rtxa** [comment]
I just find out an annoying issue with the navbar shifting to the left when the scrollbar appears or dissappear. The issue wasn't very noticeable before because the shift only affected the elements on the right side of the navbar, but now that is centered, the shifting is affecting the entire navbar. Hope someone can find a workaround for this.
![Animation](https://github.com/user-attachments/assets/442e2552-e255-4252-99b7-ea3bbff79679)

**Comment by slorber** [comment]
@rtxa the scrollbar layout shift is annoying but unfortunately I wasn't able to work around it so far. Tracked in https://github.com/facebook/docusaurus/issues/7130

**Comment by octacava** [comment]
Hi! I would like to work on this issue. I will start drafting a PR.

**Comment by slorber** [comment]
https://github.com/facebook/docusaurus/pull/12070

https://deploy-preview-12070--docusaurus-2.netlify.app/docs/

This honestly feels weird:

<img width="3826" height="1850" alt="Image" src="https://github.com/user-attachments/assets/6dc5e3bc-1f07-4552-9a10-0359d93dc444" />

With the collapsible navbar transiton, it's even more awkward:

<img width="3828" height="1852" alt="Image" src="https://github.com/user-attachments/assets/d6b2144f-01c2-4e7e-a7e4-68439a015fe5" />


I don't think this can be fixed as a standalone change; this needs a full layout redesign. More likely, we won't change the theme-classic, but introduce a new modern/redesigned theme later. 

Until then, if you have opinions about the navbar max width, this can be fixed in userland with CSS. But to me it's awkward if you don't adapt the docs sidebar position accordingly.
